src/payout/entity.rs

```diff
@@ -42,6 +45,25 @@ pub struct Payout {
     pub external_id: String,
     #[builder(setter(into), default)]
     pub metadata: Option<serde_json::Value>,
+
+    pub(super) events: EntityEvents<PayoutEvent>,
+}
+
+impl Payout {
+    pub fn cancel_payout(&mut self) {
+        self.events.push(PayoutEvent::Cancelled {
+            executed_by: self.profile_id,
```

Comment:
how do you know that it is the same profile that is canceling that created the payout? Couldn't the call be coming from another client?